mm2src/coins/utxo/utxo_common.rs

```diff
@@ -2087,18 +2157,48 @@ where
     }
 }
 
-pub fn check_all_utxo_inputs_signed_by_pub(
+/// This function is used to check that all inputs are signed/owned by the expected pubkey.
+///
+/// It's used to verify that the taker payment inputs are signed/owned by the taker's pubkey.
```

Comment:
It's actually more important in validating the dex fee transaction, this is the main use. Taker payment verification is only done by watchers.